skt/__init__.py

```diff
@@ -405,11 +405,19 @@ def merge_patchwork_patch(self, uri):
                           patchinfo.get("name").replace(',', ';')))
 
     def merge_patch_file(self, path):
+        if not os.path.isfile(path):
+            raise Exception("Patch %s not found" % os.path.basename(path))
+        abspath = os.path.abspath(path)
         try:
-            self.git_cmd("am", path)
-        except subprocess.CalledProcessError:
+            subprocess.check_output(["git", "am", abspath], cwd=self.wdir)
+        except subprocess.CalledProcessError, stdout:
             self.git_cmd("am", "--abort")
-            raise Exception("Failed to apply patch %s" % path)
+
+            with open(self.mergelog, "w") as fp:
+                fp.write(stdout.output)
+
+            raise Exception("Failed to apply patch %s" %
+                            os.path.basename(path))
```

Comment:
Same as with the nonexistent patch here, it would be better to use either original, or absolute path here rather than file name.